cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -0,0 +1,240 @@
+// no longer generates more mount points and just returns an empty output.
+func generateMountsModeInstall() error {
+	// 1. always ensure seed partition is mounted
+	isMounted, err := osutilIsMounted(filepath.Join(runMnt, "ubuntu-seed"))
+	if err != nil {
+		return err
+	}
+	if !isMounted {
+		fmt.Fprintf(stdout, "/dev/disk/by-label/ubuntu-seed %s\n", filepath.Join(runMnt, "ubuntu-seed"))
+		return nil
+	}
+	// XXX: how do we select a different recover system from the cmdline?
+
+	// 2. (auto) select recovery system for now
+	isMounted, err = osutilIsMounted(filepath.Join(runMnt, "base"))
+	if err != nil {
+		return err
+	}
+	if !isMounted {
+		seedDir, seedLabel, err := findSeed()
+		if err != nil {
+			return err
+		}
+		deviceSeed, err := seed.Open(seedDir, seedLabel)
+		if err != nil {
+			return err
+		}
+		// load assertions into a temporary database
+		if err := deviceSeed.LoadAssertions(nil, nil); err != nil {
+			return err
+		}
+		perf := timings.New(nil)
+		if err := deviceSeed.LoadMeta(perf); err != nil {
+			return err
+		}
+		// XXX: do we need more cross checks here?
+		for _, essentialSnap := range deviceSeed.EssentialSnaps() {
+			snapf, err := snap.Open(essentialSnap.Path)
+			if err != nil {
+				return err
+			}
+			info, err := snap.ReadInfoFromSnapFile(snapf, essentialSnap.SideInfo)
+			if err != nil {
+				return err
+			}
+			switch info.GetType() {
+			case snap.TypeBase:
+				fmt.Fprintf(stdout, "%s %s\n", essentialSnap.Path, filepath.Join(runMnt, "base"))
+			case snap.TypeKernel:
+				fmt.Fprintf(stdout, "%s %s\n", essentialSnap.Path, filepath.Join(runMnt, "kernel"))
+			}
+		}
+
+		return nil
+	}
+
+	// 3. mount "ubuntu-data" on a tmpfs
+	isMounted, err = osutilIsMounted(filepath.Join(runMnt, "ubuntu-data"))
+	if err != nil {
+		return err
+	}
+	if !isMounted {
+		// XXX: is there a better way?
+		fmt.Fprintf(stdout, "--type=tmpfs tmpfs /run/mnt/ubuntu-data\n")
+		return nil
+	}
+
+	// 4. write $(ubuntu_data)/var/lib/snapd/modenv
+	modeEnv := filepath.Join(runMnt, "ubuntu-data", dirs.SnapModeenvFile)
+	if !osutil.FileExists(modeEnv) {
+		if err := os.MkdirAll(filepath.Dir(modeEnv), 0755); err != nil {
+			return err
+		}
+		_, seedLabel, err := findSeed()
+		if err != nil {
+			return err
+		}
+		mode := "install"
+		modeEnvContent := fmt.Sprintf("recovery_system=%s\nmode=%s\n", seedLabel, mode)
+		if err := ioutil.WriteFile(modeEnv, []byte(modeEnvContent), 0644); err != nil {
+			return err
+		}
+	}
+
+	// 5. make ubuntu-seed available inside
+	//      ${ubuntu_data}/var/lib/snapd/seed
+	//    so that seeding in uc20 works (it expects the seeds there)
+	isMounted, err = osutilIsMounted(filepath.Join(runMnt, "ubuntu-data/var/lib/snapd/seed"))
+	if err != nil {
+		return err
+	}
+	if !isMounted {
+		seedDir := "/run/mnt/ubuntu-data/var/lib/snapd/seed"
+		if err := os.MkdirAll(seedDir, 0755); err != nil {
+			return err
+		}
+
+		// XXX: can we do this? should we --move instead?
```

Comment:
@xnox Can/should we do it this way? Happy to have a quick hangout to discuss options if not.